mitiq/zne/inference.py

```diff
@@ -38,6 +38,18 @@
 from mitiq.collector import Collector
 
 
+ExtrapolationResult = Union[
+    float,  # The zero-noise value.
+    Tuple[
+        float,  # The zero-noise value.
+        Optional[float],  # The (estimated) error on the zero-noise value.
+        List[float],  # Optimal parameters found during fitting.
+        Optional[np.ndarray],  # Covariance of fitting parameters.
+        Callable[[float], float],  # Function that was fit.
+    ],
+]
```

Comment:
Will it show up in docs?

Reply:
I think the docs would be unchanged by this PR. The return type shown in the docs would remain explicit: `Union[float Tuple[ ... ]]`.

Reply:
yeah, I meant this will not show up in docs, unlike the proposal in #651.